packages/typescript-estree/src/convert.ts

```diff
@@ -254,12 +254,8 @@ export class Converter {
     data: Omit<TSESTree.OptionalRangeAndLoc<T>, 'parent'>,
   ): T {
     const result = data;
-    if (!result.range) {
-      result.range = getRange(node, this.ast);
-    }
-    if (!result.loc) {
-      result.loc = getLocFor(result.range[0], result.range[1], this.ast);
-    }
+    result.range ??= getRange(node, this.ast);
+    result.loc ??= getLocFor(result.range[0], result.range[1], this.ast);
```

Comment:
Couldn't resist while I was here 😄 

Feels like this should be a lint rule enabled always...